Client/mods/deathmatch/logic/CPacketHandler.cpp

```diff
@@ -3176,22 +3176,20 @@ void CPacketHandler::Packet_EntityAdd(NetBitStreamInterface& bitStream)
                     SRotationDegreesSync rotationDegrees(false);
                     bitStream.Read(&rotationDegrees);
 
-                    // Read out the vehicle value as a char, then convert
-                    unsigned char ucModel = 0xFF;
-                    bitStream.Read(ucModel);
-
-                    // The server appears to subtract 400 from the vehicle id before
-                    // sending it to us, as to allow the value to fit into an unsigned
-                    // char.
-                    //
-                    // Too bad this was never documented.
-                    //
-                    // --slush
-                    unsigned short usModel = ucModel + 400;
-                    if (!CClientVehicleManager::IsValidModel(usModel))
+                    // Read out the vehicle model
+                    unsigned short usModel = 400;
+                    bitStream.Read(usModel);
+
+                    // Is invalid model -> need request model
+                    unsigned short usModelOriginal = usModel;
+                    if (!CClientVehicleManager::IsStandardModel(usModel))
                     {
-                        RaiseEntityAddError(39);
-                        return;
+                        bitStream.Read(usModelOriginal);
+                        if (!CClientVehicleManager::IsValidModel(usModel) && !g_pClientGame->m_pManager->GetModelManager()->RequestModel(usModel, usModelOriginal, eClientModelType::VEHICLE, g_pClientGame->m_pManager))
+                        {
+                            RaiseEntityAddError(39);
+                            return;
+                        }
```

Comment:
You can't simply change the value from a `char` to a `short`, this will cause packet misalignment if the server is running an older version.